google-services/Location/LocationUpdates/LocationUpdates/MainActivity.cs

```diff
@@ -1,227 +1,399 @@
+
+            mStartUpdatesButton = FindViewById<Button>(Resource.Id.start_updates_button);
+            mStopUpdatesButton = FindViewById<Button>(Resource.Id.stop_updates_button);
+            mLatitudeTextView = FindViewById<TextView>(Resource.Id.latitude_text);
+            mLongitudeTextView = FindViewById<TextView>(Resource.Id.longitude_text);
+            mLastUpdateTimeTextView = FindViewById<TextView>(Resource.Id.last_update_time_text);
+
+            mStartUpdatesButton.Click += StartUpdatesButtonHandler;
+            mStopUpdatesButton.Click += StopUpdatesButtonHandler;
+
+            mLatitudeLabel = GetString(Resource.String.latitude_label);
+            mLongitudeLabel = GetString(Resource.String.longitude_label);
+            mLastUpdateTimeLabel = GetString(Resource.String.last_update_time_label);
+
+            mRequestingLocationUpdates = false;
+            mLastUpdateTime = string.Empty;
+
+            UpdateValuesFromBundle(savedInstanceState);
+
+            mFusedLocationClient = LocationServices.GetFusedLocationProviderClient(this);
+            mSettingsClient = LocationServices.GetSettingsClient(this);
+
+            CreateLocationCallback();
+            CreateLocationRequest();
+            BuildLocationSettingsRequest();
+        }
+
+        public void UpdateValuesFromBundle(Bundle savedInstanceState)
+        {
+            if (savedInstanceState == null) return;
+            if (savedInstanceState.KeySet().Contains(KeyRequestingLocationUpdates))
+            {
+                mRequestingLocationUpdates = savedInstanceState.GetBoolean(KeyRequestingLocationUpdates);
+            }
+
+            if (savedInstanceState.KeySet().Contains(KeyLocation))
+            {
+                mCurrentLocation = (Location)savedInstanceState.GetParcelable(KeyLocation);
+            }
+
+            if (savedInstanceState.KeySet().Contains(KeyLastUpdatedTimeString))
+            {
+                mLastUpdateTime = savedInstanceState.GetString(KeyLastUpdatedTimeString);
+            }
+            UpdateUi();
+        }
+
+        public void CreateLocationRequest()
+        {
+            mLocationRequest = new LocationRequest();
+
+            mLocationRequest.SetInterval(UpdateIntervalInMilliseconds);
+
+            mLocationRequest.SetFastestInterval(FastestUpdateIntervalInMilliseconds);
+
+            mLocationRequest.SetPriority(LocationRequest.PriorityHighAccuracy);
+        }
+
+        public void CreateLocationCallback()
+        {
+            mLocationCallback = new LocationCallbackClass { Activity = this };
+        }
+
+        public void BuildLocationSettingsRequest()
+        {
+            LocationSettingsRequest.Builder builder = new LocationSettingsRequest.Builder();
+            builder.AddLocationRequest(mLocationRequest);
+            mLocationSettingsRequest = builder.Build();
+        }
+
+        protected void OnActivityResult(int requestCode, int resultCode, Intent data)
```

Comment:
We should make sure to use the correct override here, otherwise, this code will never execute. W e have mapped ResultCode to an enum as shown below, and this will also allow you to remove the `int` casting being used here.

> protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)

Reply:
Done.